src/sage/quadratic_forms/binary_qf.py

```diff
@@ -1611,7 +1612,7 @@ def solve_integer(self, n, *, algorithm="general"):
             sage: abc = [1,0,randrange(1,10^3)]
             sage: Q = BinaryQF(abc)
             sage: n = random_prime(10^9)
-            sage: if randrange(2):
+            sage: if Q[2] % 4 == 3 and randrange(2):
```

Comment:
I think the test should cover both (for random c and prime n)
Q = BinaryQF(a,b,c)
and
Q3 = BinaryQF(a, b, 4c+3)
instead of randomly testing only either Q or Q3

Reply:
Thanks, done.